app/jobs/idd2wdl.py

```diff
@@ -138,16 +149,29 @@ def target_filename(target, index):
     step_status_lock=contextlib.suppress()
   )
   step_instance.input_files_local = {input_files_local}
+
   with open(step_instance.step_status_local, "w") as status_file:
     json.dump(dict(), status_file)
+
   try:
     {count_input_reads}
     step_instance.update_status_json_file("running")
     step_instance.run()
     step_instance.count_reads()
     step_instance.save_counts()
-    step_instance.update_status_json_file("finished_running")
+    # temoorary until we instrument miniwdl - not yet uploaded, but this is the final status
+    step_instance.update_status_json_file("uploaded")
+  except idseq_dag.engine.pipeline_step.InvalidInputFileError as e:
+    try:
+        self.update_status_json_file("user_errored")
+    except:
+        logging.error("Failed to update status to 'user_errored'")
+    logging.error("Failure due to user error: %%s", traceback.format_exc())
```

Comment:
Shouldn't this (the branch under line 164) exit with an error? It looks like it falls through to a successful exit from the process.

Reply:
I don't think so. Errors due to malformed inputs from user are supposed to be pipeline successes. It means that the checks were in place and worked.
Having said we might want some kind of warning on these... so, if you feel strongly i can leave the unsuccessful exit for now - until we have a specific way to handle these somewhere eles.

Reply:
I would like us to move away from the convention that a user error results in a successful pipeline exit. A successful pipeline exit should mean that the pipeline results are available. Instead of signaling user error out-of-band, we should move to an error reporting system where the reason for the exit is communicated in the last event of the step function execution history, as described here: https://github.com/chanzuckerberg/idseq/blob/master/workflows/Interface.md#reporting-errors. The caller should then parse the reason, and if it matches a dictionary of errors attributable to user input, display a message specific to that cause.

Reply:
OK, kept the behavior. The final code is a bit convoluted but i think of for a temporary conversion.